src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -277,22 +277,18 @@ export const selectFirstAccountIdByChainId = createCachedSelector(
  * e.g. we may be swapping into a new EVM account that does not necessarily contain FOX
  * but can contain it
  */
-export const selectPortfolioAccountIdsByAssetIdFilter = createCachedSelector(
+export const selectPortfolioAccountIdsByAssetIdFilter = createDeepEqualOutputSelector(
   selectWalletAccountIds,
   selectAssetIdParamFromFilter,
-  (accountIds, assetId): AccountId[] => {
-    // early return for scenarios where assetId is not available yet
+  selectWalletId,
+  (accountIds, assetId, walletId): AccountId[] => {
+    // early return for scenarios where assetId/walletId is not available yet
     if (!assetId) return []
+    if (!walletId) return []
```

Comment:
While this may just look like an additional paranoia check, it akschually is a paranoia re: reactivity, ensuring we're not only reactive against `walletAccountIds` but against `walletId` itself.

Most likely not required, but as with everything paranoia, might as well.